rad/src/types/mod.rs

```diff
@@ -37,24 +37,23 @@ pub enum RadonTypes {
 }
 
 impl RadonTypes {
-    pub fn hash(self) -> RadResult<Hash> {
+    pub fn hash(self) -> Result<Hash, RadError> {
         self.try_into()
             .map(|vector: Vec<u8>| calculate_sha256(&*vector))
             .map(Hash::from)
-            .map_err(|_| {
-                WitnetError::from(RadError::new(
-                    RadErrorKind::Hash,
-                    String::from("Failed to hash RADON value or structure"),
-                ))
-            })
+            .map_err(|_| RadError::Hash)
     }
 }
 
 impl fmt::Display for RadonTypes {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            RadonTypes::Float(float) => write!(f, "RadonFloat: {}", float.value()),
-            _ => write!(f, "RadonType: {:?}", self),
+            RadonTypes::Array(_array) => write!(f, "RadonArray"),
+            RadonTypes::Float(_float) => write!(f, "RadonFloat"),
+            RadonTypes::Map(_map) => write!(f, "RadonMap"),
+            RadonTypes::Mixed(_mixed) => write!(f, "RadonMixed"),
+            RadonTypes::String(_string) => write!(f, "RadonString"),
+            //_ => write!(f, "RadonTypes::{:?}", self),
```

Comment:
The display implementation should show some information about the inner value, not only the type.